iwyu_globals.cc

```diff
@@ -308,6 +313,13 @@ int CommandlineFlags::ParseArgv(int argc, char** argv) {
           exit(EXIT_FAILURE);
         }
         break;
+      case 'p':
+        if (strcmp(optarg, "clang_mappings") == 0) {
```

Comment:
Could you model this after `--debug` instead, and store a set of flag strings? Seems nice to use the same design for both.